app/routes/github.webhook/pull-request-webhook.ts

```diff
@@ -13,25 +13,65 @@ export async function processPullRequestWebhook(
   repository: Repository
 ) {
   if (!repository.deployOnlyOnPullRequest)
-    return json({ message: "No action taken" });
+    return json({
+      status: "skipped",
+      message: "No action taken",
+      reason: "Repository is not configured for PR-only deployments",
+      repository: repository.id
+    });
 
   const [files, branch] = await findPullRequestData();
 
-  if (!branch && !hasChangesAffectingNextjsApp())
-    return json({ message: "No action taken" });
+  if (!branch) {
```

Comment:
This introduces a bug...  this now means "if we've never deployed before, don't deploy" and thus no deploys will happen.

I think we should leave the `if` the way it was, but conditionally set the response message.

Reply:
Just commented. I see your point.